crates/floresta-wire/src/p2p_wire/node.rs

```diff
@@ -241,14 +241,30 @@ where
             initial_height: peer.height,
         })
     }
-    pub(crate) fn handle_disconnection(&mut self, peer: u32, idx: usize) -> Result<(), WireError> {
+    pub(crate) async fn handle_disconnection(
```

Comment:
I guess, we have to make `handle_disconnect` async now.

Reply:
Yes, that's fine.